lightning/src/ln/channelmanager.rs

```diff
@@ -1828,30 +1845,16 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			return Err(APIError::APIMisuseError { err: format!("Channel value must be at least 1000 satoshis. It was {}", channel_value_satoshis) });
 		}
 
-		let height = self.best_block.read().unwrap().height();
-		let mut outbound_scid_alias = 0;
-		let mut i = 0;
-		loop {
-			#[cfg(fuzzing)] { // fuzzing chacha20 doesn't use the key at all so we always get the same alias
-				outbound_scid_alias += 1;
-			}
-			#[cfg(not(fuzzing))] {
-				outbound_scid_alias = fake_scid::Namespace::OutboundAlias.get_fake_scid(height, &self.genesis_hash, &self.fake_scid_rand_bytes, &self.keys_manager);
-			}
-			if self.outbound_scid_aliases.lock().unwrap().insert(outbound_scid_alias) { break; }
-			i += 1;
-			if i > 1_000_000 { panic!("Your RNG is busted or we ran out of possible outbound SCID aliases (which should never happen before we run out of memory to store channels"); }
-		}
-
 		let channel = {
 			let per_peer_state = self.per_peer_state.read().unwrap();
 			match per_peer_state.get(&their_network_key) {
 				Some(peer_state) => {
+					let outbound_scid_alias = self.create_insert_outbound_scid_alias();
 					let peer_state = peer_state.lock().unwrap();
 					let their_features = &peer_state.latest_features;
 					let config = if override_config.is_some() { override_config.as_ref().unwrap() } else { &self.default_configuration };
 					Channel::new_outbound(&self.fee_estimator, &self.keys_manager, their_network_key, their_features,
-						channel_value_satoshis, push_msat, user_channel_id, config, height, outbound_scid_alias)?
+						channel_value_satoshis, push_msat, user_channel_id, config, self.best_block.read().unwrap().height(), outbound_scid_alias)?
```

Comment:
I don't think I see any lock ordering concerns between `per_peer_state` and `best_block`, but perhaps docs could mention ordering.

Reply:
Even better, let me just go update the debug lock stuff to support RwLock, will follow up with a separate PR.

Reply:
#1352